Source/JavaScriptCore/runtime/JSGlobalObject.cpp

```diff
@@ -2550,6 +2570,13 @@ JSGlobalObject* JSGlobalObject::create(VM& vm, Structure* structure)
     return globalObject;
 }
 
+JSGlobalObject* JSGlobalObject::create(VM& vm, Structure* structure, const GlobalObjectMethodTable* methodTable)
+{
+    JSGlobalObject* globalObject = new (NotNull, allocateCell<JSGlobalObject>(vm.heap)) JSGlobalObject(vm, structure, methodTable);
+    globalObject->finishCreation(vm);
+    return globalObject;
+}
```

Comment:
this was the best way I could think of creating a new global object with a custom method table. We grab the method table from the original global object in https://github.com/WebKit/WebKit/pull/8/files#diff-44ebd0942590c3af4f75c8833b1b8f2dd9d649fd9e7dcc7a9e1410df5711c22fR33, which might be defined in `jsc.cpp` or elsewhere in webkit when run in the browser context.